mautic/mautic
# DPMMA-902 Campaign priority

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | https://github.com/mautic/user-documentation/pull/200
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds possibility to set priority of campaign (normal / high). Campaigns with higher priority will be processed before campaigns with normal priority when running the cron job.
![image](https://github.com/mautic/mautic/assets/8580942/c9e47a86-91da-448a-b2ca-00dba9c66e1b)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add two campaigns with default (normal) priority
3. Run `php bin/console m:c:t` and check the order  - the oldest campaign should be executed first
4. Change the priority of the second campaign to high
5. Again run `php bin/console m:c:t` - now the second campaign should be executed first
